### ISSUE TITLE
pg-pool Callback Error at pg-pool/index.js:376

### ISSUE THREAD

**Issue by mhstoller** [brianc/node-postgres#2570]
```
/***/node_modules/pg-pool/index.js:376
          return cb(err)
                 ^

TypeError: cb is not a function
    at Query.callback (/***/node_modules/pg-pool/index.js:376:18)
    at Query.handleError (***/node_modules/pg/lib/query.js:128:19)
    at /***/node_modules/pg/lib/client.js:484:30
    at processTicksAndRejections (internal/process/task_queues.js:75:11)
```

Caused by not providing an array of arguments, and instead making a typo like such:
`await pool.query(query, param, param2);`

**Comment by luizgribeiro** [comment]
Hey @mhstoller!

**TLDR:** It's not an issue. The query method can have 1 (query), 2 (array values) or 3 (callback) parameters. 

I couldn't understand the reason why it's an issue. 

As presented in the [pg-poll doc](https://node-postgres.com/api/pool) (bellow) you can pool.query passing 3 arguments to it:

- The query (string/text)
- An array of values (sanitized and interpolated on the quey)
- A callback function that has arguments `error` and `result`

Since you are using await, what you probably want is the promisified version of this function. The thing is that the implemented function does not have a signature in which it receives a query and its parameters values as a variable number of parameters.
More than that, the current approach to this implementation would not work. JavaScript does not support function overload by signature (on this case, number of parameters). 

Have a look on the following code, that implements the `query` function:

```javascript  
query(text, values, cb) {
    // guard clause against passing a function as the first parameter
    if (typeof text === 'function') {
      const response = promisify(this.Promise, text)
      setImmediate(function () {
        return response.callback(new Error('Passing a function as the first parameter to pool.query is not supported'))
      })
      return response.result
    }

    // allow plain text query without values
    if (typeof values === 'function') {
      cb = values
      values = undefined
    }
    const response = promisify(this.Promise, cb)
    cb = response.callback

    this.connect((err, client) => {
      if (err) {
        return cb(err)
      }

      let clientReleased = false
      const onError = (err) => {
        if (clientReleased) {
          return
        }
        clientReleased = true
        client.release(err)
        cb(err)
      }

      client.once('error', onError)
      this.log('dispatching query')
      client.query(text, values, (err, res) => {
        this.log('query dispatched')
        client.removeListener('error', onError)
        if (clientReleased) {
          return
        }
        clientReleased = true
        client.release(err)
        if (err) {
          return cb(err)
        } else {
          return cb(undefined, res)
        }
      })
    })
    return response.result
  }

  end(cb) {
    this.log('ending')
    if (this.ending) {
      const err = new Error('Called end on pool more than once')
      return cb ? cb(err) : this.Promise.reject(err)
    }
    this.ending = true
    const promised = promisify(this.Promise, cb)
    this._endCallback = promised.callback
    this._pulseQueue()
    return promised.result
  }
```

Perhaps you've seen somenthing similar to this being done in another lib, but it won't work with pg.

**Comment by mhstoller** [comment]
Ahh, so the error is actually that "param2" is not a function as pg expects it to be a callback function. Thank you for pointing this out to me!! I thought there was an error within pg not being able to execute a callback, but it's just my crappy code ;)

Closing.
